modules/widgets/src/compass-widget.tsx

```diff
@@ -0,0 +1,118 @@
+import {FlyToInterpolator, WebMercatorViewport, _GlobeViewport} from '@deck.gl/core';
+import type {Deck, Viewport, Widget, WidgetPlacement} from '@deck.gl/core';
+import {h, render} from 'preact';
+
+interface CompassWidgetProps {
+  id: string;
+  viewId?: string | null;
+  placement?: WidgetPlacement;
+  CompassLabel?: string;
+  transitionDuration?: number;
+  style?: Partial<CSSStyleDeclaration>;
+}
+
+const ICON_HEIGHT = 16;
```

Comment:
How can we enable scaling the icon with css in the use case where the user is changing the size for, say, mobile?

The rest of the button size with `--button-size`

Reply:
This size was only used for the viewBox, the icon can still be resized independently. But I removed the constant and hardcoded the values. I think it was not very useful, but rather confusing.